### PR TITLE
chore: use GitHub actions to build and publish libraries

### PR DESCRIPTION
**Pull request by Shobhajayanna**
Replace Jenkins with GitHub actions as a tool for automation.

### CODE REVIEW

**Comment by pablocarle** [comment]
Which snapshot version was published?

**Comment by pablocarle** [comment]
Can you update the settings to only allow merging with approving?

**Comment by pablocarle** [comment]
Should it already appear in https://zowe.jfrog.io/artifactory/list/libs-snapshot-local/org/zowe/apiml/sdk/common-java-lib-package ?

File: .github/actions/native-build/action.yml [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+    required: true
+  passwd:
+    description: "SSH password"
+    required: true
+runs:
+  using: "composite"
+  steps:
+    - name: install zowe/cli
+      shell: bash
+      run: npm install -g @zowe/cli@6.32.2
```
Comment (pablocarle) [inline .github/actions/native-build/action.yml]: can you update this one to v1 lts?

Comment (pablocarle) [reply .github/actions/native-build/action.yml]: anyway eventually we will need to update it to v2 team config files

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+
+      - name: Use Node LTS
+        uses: actions/setup-node@v3
+        with:
+          node-version: 'lts/*'
+      - name: install zowe/cli
+        run: npm install -g @zowe/cli@6.32.2
```
Comment (pablocarle) [inline .github/workflows/release.yml]: also here, v1 lts

File: .github/workflows/release.yml [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+      - name: install zowe-api-dev
+        run: npm -g install @zowedev/zowe-api-dev
+
+      - name: Release or snapshot
+        run: |
+          if [ ${{github.event.inputs.snapshot_version }} ];
+          ./gradlew publish -x test -Pzowe.deploy.username=$ARTIFACTORY_USERNAME -Pzowe.deploy.password=$ARTIFACTORY_PASSWORD -Partifactory_user=$ARTIFACTORY_USERNAME -Partifactory_password=$ARTIFACTORY_PASSWORD
+          else
+          ./gradlew release -x test -Prelease.useAutomaticVersion=true -Pzowe.deploy.username=$ARTIFACTORY_USERNAME -Pzowe.deploy.password=$ARTIFACTORY_PASSWORD -Partifactory_user=$ARTIFACTORY_USERNAME -Partifactory_password=$ARTIFACTORY_PASSWORD
```
Comment (pablocarle) [inline .github/workflows/release.yml]: how is this automatic version going to work?

Comment (achmelo) [reply .github/workflows/release.yml]: it was not done properly, I have split it into release and publish, release will be done by initiating the workflow manualy

**Comment by achmelo** [comment]
> Should it already appear in https://zowe.jfrog.io/artifactory/list/libs-snapshot-local/org/zowe/apiml/sdk/common-java-lib-package ?

now it is there https://zowe.jfrog.io/artifactory/libs-snapshot-local/org/zowe/apiml/sdk/common-java-lib-package/2.0.1-SNAPSHOT/

**Comment by achmelo** [comment]
> Which snapshot version was published?

the one from gradle properties, do you think we need an automatic update of the snapshot version, currently release task should update it

**Comment by achmelo** [comment]
> Can you update the settings to only allow merging with approving?

done

File: common-java-lib-package/build.gradle [hunk 4]
```diff
@@ -41,8 +41,8 @@ packageCommonLib.doFirst {
     getCommitHash.waitForOrKill(1000)
 
     // BRANCH_NAME and BUILD_NUMBER are Jenkins variables
```
Comment (pablocarle) [inline common-java-lib-package/build.gradle]: we can probably delete (or update) this comment

**Review by pablocarle — APPROVED** [review]
Thank you for addressing my comments

**Pull request merged**